cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/query_enhancements/saved_search.spec.js

```diff
@@ -0,0 +1,513 @@
+    cy.getElementByTestId('datasetSelectorNext').click();
+
+    cy.getElementByTestId('advancedSelectorTimeFieldSelect').select('timestamp');
+    cy.getElementByTestId('advancedSelectorConfirmButton').click();
+
+    cy.getElementByTestId('datasetSelectorButton').should(
+      'contain.text',
+      `${datasourceName}::${dataset}`
+    );
+  }
+};
+
+const setDatePickerDatesAndSearchIfRelevant = (language) => {
+  if (language === 'OpenSearch SQL') {
+    return;
+  }
+
+  cy.setTopNavDate(START_TIME, END_TIME);
+};
+
+const setSearchConfigurations = ({
+  addFilter,
+  queryString,
+  setHistogramInterval,
+  selectFields,
+  applySort,
+}) => {
+  if (addFilter) {
+    cy.getElementByTestId('showFilterActions').click();
+    cy.getElementByTestId('addFilters').click();
+    cy.getElementByTestId('filterFieldSuggestionList').find('input').type('category');
+    cy.getElementByTestId('comboBoxOptionsList filterFieldSuggestionList-optionsList')
+      .find('button[title="category"]')
+      .click();
+    cy.getElementByTestId('filterOperatorList').find('input').type('is one of');
+    cy.getElementByTestId('comboBoxOptionsList filterOperatorList-optionsList')
+      .find('button[title="is one of"]')
+      .click();
+    cy.getElementByTestId('filterParams').find('input').type('Application');
+    cy.getElementByTestId(
+      'comboBoxOptionsList filterParamsComboBox phrasesParamsComboxBox-optionsList'
+    )
+      .find('button[title="Application"]')
+      .click();
+    // Need to wait here a bit to avoid cypress flakiness
+    cy.wait(750);
+    cy.get('span[title="Application"]').should('be.visible');
+    // Need to wait here a bit to avoid cypress error
+    cy.wait(750);
+    // force is true below because sometimes a dropdown covers the button
+    cy.getElementByTestId('saveFilter').click({ force: true });
+  }
+
+  cy.setQueryEditor(queryString);
+
+  if (setHistogramInterval) {
+    cy.getElementByTestId('discoverIntervalSelect').select('w');
+  }
+
+  if (selectFields) {
+    cy.getElementByTestId('fieldToggle-bytes_transferred').click();
+    cy.getElementByTestId('fieldToggle-personal.name').click();
+
+    // reloading as field filtering doesn't appear right away on cypress. Issue only appears in cypress tests,
+    // so resolving it via a force reload.
+    // cy.reload();
+    cy.getElementByTestId('querySubmitButton').should('be.visible');
+  }
+
+  if (applySort) {
+    cy.getElementByTestId('docTableHeaderFieldSort_bytes_transferred').click();
+
+    // stop sorting based on timestamp
+    cy.getElementByTestId('docTableHeaderFieldSort_timestamp').click();
+    cy.getElementByTestId('docTableHeaderFieldSort_timestamp').trigger('mouseover');
+    cy.contains('div', 'Sort timestamp ascending').should('be.visible');
+
+    cy.getElementByTestId('docTableHeaderFieldSort_bytes_transferred').click();
+
+    // TODO: This reload shouldn't need to be here, but currently the sort doesn't always happen right away
```

Comment:
I don't quite get it. For DQL, sort should happen after waitForLoad. Do you mind adding a video for this?

Reply:
I can't always replicate it, but sometimes:

- The "Selected fields" are not applied. The headers for the selected fields are there, but all the data is dumped into the first column
- The Sort sometimes doesn't happen, even after loading for a while

I'll grab videos of these if i can replicate it

Reply:
I got a video of the latter case, but its too big to upload here :D . I'll share it with you in private